driver-sync/src/test/functional/com/mongodb/client/ClientSideEncryptionViewAreProhibitedTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client;
+
+import com.mongodb.AutoEncryptionSettings;
+import com.mongodb.MongoClientSettings;
+import com.mongodb.MongoException;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.UuidRepresentation;
+import org.bson.codecs.UuidCodec;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.mongodb.ClusterFixture.isNotAtLeastJava8;
+import static com.mongodb.ClusterFixture.serverVersionAtLeast;
+import static junit.framework.TestCase.assertTrue;
+import static com.mongodb.client.Fixture.getMongoClientSettingsBuilder;
+import static org.bson.codecs.configuration.CodecRegistries.fromCodecs;
+import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+import static org.junit.Assume.assumeFalse;
+import static org.junit.Assume.assumeTrue;
+
+public class ClientSideEncryptionViewAreProhibitedTest {
+    private MongoClient client, clientEncrypted;
+
+    @Before
+    public void setUp() {
+        assumeFalse(isNotAtLeastJava8());
+        assumeTrue(serverVersionAtLeast(4, 1));
+        assumeTrue("Encryption test with external keyVault is disabled",
+                System.getProperty("org.mongodb.test.awsAccessKeyId") != null
+                        && !System.getProperty("org.mongodb.test.awsAccessKeyId").isEmpty());
+
+        MongoClientSettings.Builder clientSettingsBuilder = getMongoClientSettingsBuilder();
+        CodecRegistry codecRegistry = fromRegistries(
```

Comment:
Where did you get this from?  I don't think it's necessary to alter the UuidCodec if we're always using BsonDocument.  Try removing it and see if the tests still pass.  If so, I'm not sure if you need a new MongoClient for this test at all.  You can just grab the global one from the Fixture.